### DISCUSSION TITLE
Missing data data when Providers.Credentials is setup

### DISCUSSION THREAD

**Discussion by IRediTOTO** [nextauthjs/next-auth#1439] [Help]
**Describe the bug**
Hi I define option of next-auth like this
```js
const options = {
    providers: [
        Providers.Google({
            clientId: process.env.GOOGLE_CLIENT_ID,
            clientSecret: process.env.GOOGLE_CLIENT_SECRET
        }),
        Providers.Credentials({
            name: 'Credentials',
            credentials: {
                email: { label: "Email", type: "text", placeholder: "jsmith" },
                password: { label: "Password", type: "password" }
            },
            async authorize(credentials) {
                const db = await mongoDb()
                // Add logic here to look up the user from the credentials supplied
                const user = await db.collection("users").findOne({
                    email: credentials.email,
                    password: credentials.password
                })

                if (user) {
                    // Any object returned will be saved in `user` property of the JWT
                    return user
                } else {
                    // If you return null or false then the credentials will be rejected
                    return null
                    // You can also Reject this callback with an Error or with a URL:
                    // throw new Error('error message') // Redirect to error page
                    // throw '/path/to/redirect'        // Redirect to a URL
                }
            }
        })
    ],
    database: process.env.MONGODB_URI,
    callbacks: {
        session: async (session, user) => {
            session.user.myCustomData = user.myCustomData
            return Promise.resolve(session)
        },
      
    },
    adapter: Adapters.TypeORM.Adapter(
        process.env.MONGODB_URI,
        {
            models: {
                User: models.User,
            },
        }
    ),
    session: {
        jwt: true,
    },
}
```
I use custom Users model
```
export const UserSchema = {
  name: "User",
  target: User,
  columns: {
    ...Adapters.TypeORM.Models.User.schema.columns,
    // Adds a phoneNumber to the User schema
    myCustomData: {
      type: "varchar",
      nullable: true,
    },
  },
}
```
Everything work well, user can login with normal but the `session =useSesstion()` don't return `myCustomData `anymore
```
{"user":
"name":"Admin",
"email":"abc@gmail.com",
"image":"https://abc.com/ssdf.png",
"accessToken":"5856be531fa8a5a07462ae48af6e0cc326db39840d07b249c7466fdffb319285",
"expires":"2021-04-03T23:10:32.937Z"}
```

I tried to remove  Providers.Credentials and turn off jwt I will get the custom data I need.
```
{"user":
"name":"Admin",
"email":"abc@gmail.com",
"image":"https://abc.com/ssdf.png",
"myCustomData":"Hello world"},
"accessToken":"5856be531fa8a5a07462ae48af6e0cc326db39840d07b249c7466fdffb319285",
"expires":"2021-04-03T23:10:32.937Z"}
```

**Steps to reproduce**
I think don't need it, i just trying learn how to use this package

**Expected behavior**
I need more data in `useSesstion` hook, the user can have more data like user's credit, user's bought subscription, ratting,... which will show in header. I don't want query database again for optimize

**Screenshots or error logs**
No screenshot, this is not an error, its something missing

- [ ]  Found the documentation helpful

* [ ] Found documentation but was incomplete
* [ ] Could not find relevant documentation
* [ ] Found the example project helpful
* [ ] Did not find the example project helpful

**Comment by StefanSelfTaught** [comment]
Shouldn't the user queried in authorize function should contain the `myCustomData` property?

**Reply by IRediTOTO** [reply]
Yes, the user in that func contain all fiels of that user. But it just show name, email and image in useSession(). 
I want control what data will be returned in useSession() and getSession() on server.

**Reply by balazsorban44** [reply]
You can totally do that, look at the jwt and session callbacks: https://next-auth.js.org/configuration/callbacks

By default, we only return a subset of the user data to be secure.

**Reply by IRediTOTO** [reply]
Yes, i did it

callbacks: {
        session: async (session, user) => {
            session.user.myCustomData = user.myCustomData
            return Promise.resolve(session)
        },
      
    },
But it not show in useSession() thats the problem.

**Reply by balazsorban44** [reply]
check out the jwt callback as well. 😊

https://next-auth.js.org/configuration/callbacks#jwt-callback

**Comment by mhdalmajid** [comment]
this example will help you 

```javascript 
const callbacks = {
  async jwt(token, user) {
    if (user) {
      token.accessToken = user.token
      token.name = user.userName
      token.picture = user.avatar
    }
    return token
  },

  async session(session, user) {
    session.accessToken = user.accessToken
    return session
  },

}
```
